File: backend/src/main/java/com/crud/productos/entity/Producto.java
package com.crud.productos.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Id;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Column;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "productos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String nombre;

    private String marca;

    private String categoria;

    @Column(nullable = false)
    private Double precio;

    @Column(nullable = false)
    private Integer existencias;

    @Column(nullable = false, columnDefinition = "boolean default true")
    private Boolean activo = true;
}

File: frontend/src/app/models/producto.ts
export interface Producto {
  id?: number;
  nombre: string;
  marca?: string;
  categoria?: string;
  precio: number;
  existencias: number;
  activo?: boolean;
}

export interface AjusteInventario {
  ajuste: number;
  justificacion: string;
}

export interface PaginatedResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}
